packages/relay/src/lib/eth.ts

```diff
@@ -620,13 +627,18 @@ export class EthImpl implements Eth {
     if (blockNumber === 0) {
       return '0x0';
     } else {
-      const result = await this.mirrorNodeClient.resolveEntityType(address);
-      if (result?.type === constants.TYPE_ACCOUNT) {
-        const accountInfo = await this.sdkClient.getAccountInfo(result?.entity.account, EthImpl.ethGetTransactionCount);
-        return EthImpl.numberTo0x(Number(accountInfo.ethereumNonce));
-      }
-      else if (result?.type === constants.TYPE_CONTRACT) {
-        return EthImpl.numberTo0x(1);
+      try {
+        const result = await this.mirrorNodeClient.resolveEntityType(address);
+        if (result?.type === constants.TYPE_ACCOUNT) {
+            const accountInfo = await this.sdkClient.getAccountInfo(result?.entity.account, EthImpl.ethGetTransactionCount);
+            return EthImpl.numberTo0x(Number(accountInfo.ethereumNonce));  
+        }
+        else if (result?.type === constants.TYPE_CONTRACT) {
+          return EthImpl.numberTo0x(1);
+        }  
+      } catch (e: any) {
+        this.logger.error(e, 'Error raised during getTransactionCount for address %s, block number or tag %s', address, blockNumOrTag);
+        throw e;        
```

Comment:
All returns should be of predefined types where possible
```suggestion
        return predefined.INTERNAL_ERROR;        
```